rules/rbe_repo.bzl

```diff
@@ -0,0 +1,597 @@
+                   "you can use this attr to indicate a type of config (e.g., default, " +
+                   "msan). The configs will be generated in a sub-directory when this attr  " +
+                   "is used."),
+        ),
+        "setup_cmd": attr.string(
+            default = "cd .",
+            doc = ("Optional. Pass an additional command that will be executed " +
+                   "(inside the container) before running bazel to generate the " +
+                   "toolchain configs"),
+        ),
+        "revision": attr.string(
+            mandatory = True,
+            doc = ("The revision of the rbe-ubuntu16-04 container"),
+        ),
+    },
+    environ = [
+        RBE_AUTOCONF_ROOT,
+    ],
+    implementation = _impl,
+)
+
+load("@bazel_toolchains//rules:environments.bzl", "clang_env")
+
+def rbe_autoconfig(
+        name,
+        bazel_version = None,
+        bazel_rc = None,
+        output_base = "",
+        config_dir = "",
+        revision = "latest",
+        env = clang_env()):
+    """ Creates a repository with toolchain configs generated for an rbe-ubuntu container.
+
+    This macro wraps (and simplifies) invocation of _rbe_autoconfig rule.
+    Use this macro in your WORKSPACE.
+
+    Args:
+      bazel_version: The version of Bazel to use to generate toolchain configs.
+          `Use only (major, minor, patch), e.g., '0.20.0'.
+      bazel_rc: The rc (for the given version of Bazel) to use. Must be published
+          in https://releases.bazel.build
+      output_base: Optional. The directory (under the project root) where the
+          produced toolchain configs will be copied to.
+      config_dir: Optional. Subdirectory where configs will be copied to.
+          Use only if output_base is defined.
+      revision: a revision of an rbe-ubuntu16-04 container.
+          See gcr.io/cloud-marketplace/google/rbe-ubuntu16-04
+      env: dict. Additional env variables that will be set when running the
+          Bazel command to generate the toolchain configs.
+    """
+    if output_base == "" and config_dir != "":
+        fail("config_dir can only be used when output_base is set.")
+    if revision == "latest":
+        revision = RBE_UBUNTU16_04_LATEST
+    if bazel_rc and not bazel_version:
+        fail("bazel_rc can only be used with bazel_version.")
+    digest = public_rbe_ubuntu16_04_sha256s().get(revision, None)
+    if not digest:
+        fail(("Could not find a valid digest for revision %s " +
+              "please check it is declared in " +
+              "@bazel_toolchains//rules:toolchain_containers.bzl" % revision))
+    _rbe_autoconfig(
+        name = name,
+        bazel_version = bazel_version,
+        bazel_rc = bazel_rc,
+        config_dir = config_dir,
+        digest = digest,
+        env = env,
+        output_base = output_base,
+        revision = revision,
+        # We use 'l.gcr.io' to not require users to do gcloud login
+        registry = "l.gcr.io",
```

Comment:
Ok I see what image is being pulled here. So this autoconfig only works for ubuntu then? Would it make sense to expose this to allow users to generate bazel configs on other operating systems?

Reply:
I've significantly simplified the process of pulling and unpacking an image in a way that can only be guaranteed to work with our containers (i.e., though it might work with other containers, I cannot guarantee it atm, so I'm not exposing ways of overriding the container to use, or the config repos to pull from the container. For those use cases the regular build rule (docker_autoconfig) should still be the recommended path (and it works well for them).